src/cmd/MVP-4/main.go

```diff
@@ -110,62 +73,78 @@ func main() {
-			logger.Update(values)
-			if idToType[id] == "data" {
-				dataTransfer.Update(packetFactory.NewPacketUpdate(id, packet, values))
+		for update := range updateChan {
+			logger.Update(update.Fields)
+			if idToType[update.ID] == "data" {
+				dataTransfer.Update(update)
 			} else {
-				messageTransfer.Broadcast(packetFactory.NewPacketUpdate(id, packet, values))
+				messageTransfer.Broadcast(update)
 			}
 		}
 	}(messageChannel)
 
 	go func() {
 		for order := range orderChannel {
-			order.Values = displayConverter.Revert(order.Values)
-			order.Values = podConverter.Revert(order.Values)
-			transportController.Write(idToIP[order.ID], packetParser.Encode(order.ID, order.Values))
+			log.Println(order)
+			if err := boardMux.Request(order); err != nil {
+				log.Printf("request failed: %s\n", err)
+			}
 		}
 	}()
 
 	httpServer := server.Server{Router: mux.NewRouter()}
-	httpServer.ServeData(os.Getenv("POD_DATA_ENDPOINT"), podData)
-	httpServer.ServeData(os.Getenv("ORDER_DATA_ENDPOINT"), orderData)
 
-	handle := websocket_handle.RunWSHandle(httpServer.Router, "/backend", map[string]chan models.MessageTarget{
+	httpServer.ServeData("/backend/"+os.Getenv("POD_DATA_ENDPOINT"), podData)
+	httpServer.ServeData("/backend/"+os.Getenv("ORDER_DATA_ENDPOINT"), orderData)
+
+	websocket_handle.RunWSHandle(httpServer.Router, "/backend", map[string]chan models.MessageTarget{
 		"podData/update":    dataTransferChannel,
 		"message/update":    messageChannel,
-		"order/update":      ordChannel,
+		"order/send":        ordChannel,
 		"connection/update": connectionChannel,
-		"logger/enable":     loggerChannel,
+		"logger":            loggerChannel,
 	})
 
 	path, _ := os.Getwd()
 	httpServer.FileServer("/", filepath.Join(path, "static"))
 
 	go httpServer.ListenAndServe(os.Getenv("SERVER_ADDR"))
 
-	//Test with clients "staticClient"
-	httpServerClient := server.Server{Router: mux.NewRouter()}
-	httpServerClient.FileServer("/", filepath.Join(path, "staticClient"))
-
-	go httpServerClient.ListenAndServe(os.Getenv("SERVER_CLIENT_ADDR"))
-
 	interrupt := make(chan os.Signal, 1)
 	signal.Notify(interrupt, os.Interrupt)
 
+	log.Println("backend running!")
 loop:
 	for {
 		select {
 		case <-time.After(time.Second * 10):
-			fmt.Println(transportController.Stats())
+			fmt.Println(vehicle.Stats())
 		case <-interrupt:
 			break loop
 		}
 	}
+}
 
-	log.Println(handle, orderTransfer)
+func getFPS(fps int) time.Duration {
+	return time.Duration(int(time.Second) / fps)
+}
+
+func getIdToType(podData *vehicle_models.PodData) map[uint16]string {
+	idToType := make(map[uint16]string)
+	for _, brd := range podData.Boards {
+		for _, pkt := range brd.Packets {
+			idToType[pkt.ID] = "data"
+		measurements_loop:
+			for msr := range pkt.Measurements {
+				if msr == "warning" {
+					idToType[pkt.ID] = "warning"
+					break measurements_loop
+				} else if msr == "fault" {
+					idToType[pkt.ID] = "fault"
+					break measurements_loop
+				}
+			}
```

Comment:
may be too much for a function

Reply:
Of course, use breaks is a clear symptom. At least 3 methods should be extracted from here